src/XMakeTasks/UnitTests/Hash_Tests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.UnitTests;
+using Microsoft.Build.Utilities;
+using Xunit;
+
+namespace Microsoft.Build.Tasks.UnitTests
+{
+    public class Hash_Tests
+    {
+        [Fact]
+        public void HashTaskTest()
+        {
+            // This hash was pre-computed. If the implementation changes it may need to be adjusted.
+            var expectedHash = "57eafbc4c4ec405456e02834107118f1dc67e73f";
+
+            var actualHash = ExecuteHashTask(new ITaskItem[]
+            {
+                new TaskItem("Item1"), new TaskItem("Item2"), new TaskItem("Item3")
+            });
+            Assert.Equal(expectedHash, actualHash);
+
+            // Try again to ensure the same hash
+            var actualHash2 = ExecuteHashTask(new ITaskItem[]
+            {
+                new TaskItem("Item1"), new TaskItem("Item2"), new TaskItem("Item3")
+            });
+            Assert.Equal(expectedHash, actualHash2);
+        }
+
+        [Fact]
+        public void HashTaskEmptyInputTest()
+        {
+            System.Diagnostics.Debugger.Launch();
+            System.Diagnostics.Debugger.Break();
+
+            // Hash should be valid for empty item
+            var emptyItemHash = ExecuteHashTask(new ITaskItem[] {new TaskItem("")});
```

Comment:
There could be another test for an empty array: ExecuteHashTask(new ITaskItem[0]);